tests/trafficmanagement/ingress/secure_ingress_sds_test.go

```diff
@@ -42,6 +42,7 @@ func TestSecureIngressSDS(t *testing.T) {
 	// Check the version of curl. This test requires the --retry-connrefused arg.
 	curl.RequireMinVersionOrFail(t, semver.MustParse("7.52.0"))
 
+	t.Skip("Skipping mTLS")
```

Comment:
Why skip this one? It's a little flaky, but does pass most of the time.

Reply:
I was testing a theory that some of the tests were enabling mTLS in strict mode, which was causing this test to fail because the configuration was sticking around between tests.  Based on my local testing and this run, it looks like that's the case:
```
2020-05-08T05:32:21.937644Z	info	ci	Running command script scripts/authz_tcp.txt
2020-05-08T05:32:36.257556Z	info	ci	Creating snippets for input: scripts/authz_tcp.txt
2020-05-08T05:32:36.257646Z	info	ci	Running command script cleanup.sh
2020-05-08T05:33:20.376007Z	info	ci	Creating snippets for input: cleanup.sh
2020-05-08T05:33:20.376104Z	info	ci	=== DONE (passed):  Test: 'authz_tcp[TestAuthzTCP] (2m9.343649647s)' ===
--- PASS: TestAuthzTCP (129.34s)
```
(This test run failing was from a different test, not TestAuthzTCP.)

I was under the impression that the environment got cleaned up between runs.  (I thought this was brought up as one of the reasons that running the tests were slow.)

There are a couple of issues that I see:
    1) Should we be cleaning up the state between these tests?  (I'd think "yes", since it could lead to subtle cascading failures like this one.  However, we'll need to keep runtime in mind.)
    2) Why is this test failing with mTLS enabled?  On my team, there seems to be debate about whether this is expected behavior.  If this is expected behavior, in my opinion, we should update the task to handle this situation, since even though the task doesn't say to enable mTLS, someone could have mTLS enabled and be playing with these tasks, which would lead to confusion.

I'm still doing some investigation on my end about why this doesn't work with mTLS strict enabled.

Reply:
I'm not suggesting that we disable mTLS in the tests.  😄

Reply:
Thanks for the explanation.

1. Yes, the tests are supposed to cleanup when they're finished, but there may be tests that are forgetting something in their cleanup script. Maybe some test is just doing what the task document says, which has something missing. Tasks that create their own namespace, and then delete the namespace at cleanup time are less likely to have an issue than ones that use the default namespace, or if they set global config. It sounds like that might be what you've found here. Only question is which test is doing it?
2. I don't know either if the TCP echo is supposed to work the same with or without mTLS, but I would think it should. If not, I guess what you're suggesting is that there should be another step in the Before you begin section to disable mTLS? That would make sense to me too.

Anyway, thanks for digging into this.